packages/grafana-ui/src/types/datasource.ts

```diff
@@ -437,6 +444,29 @@ export interface ScopedVars {
   [key: string]: ScopedVar;
 }
 
+enum StreamingResponseShape {
+  /**
+   * Full DataFrames
+   */
+  full = 'full',
+
+  /**
+   * Send only the changes since the last notifiction
+   */
+  delta = 'delta',
+}
+
+interface ObserverConfig {
+  pause?: boolean;
+  throttle?: number;
+}
+
+export interface StramingQueryOptions {
+  frames?: StreamingResponseShape;
+  listener?: ObserverConfig;
+  notification?: ObserverConfig;
```

Comment:
Honestly no idea how this should work. If I want a stream of data from datasource I call `query` function with `DataQueryRequest` as an argument. Does it make sense to send an option that says `pause: true` what will that do? If I want to pause it will I call `query()` function again or what?

Reply:
So do you mean DataSource would keep track of query -> stream and do some kind of deduplication? And that by giving the same query but with different options you could do action on that query stream?

Reply:
I am again wondering if that is not too much to put into a DS. I mean if we had reusable query model and suitable UI to that this would not be a huge problem I think and would leave the DS implementation relatively naive and easy. Even if we say this is a requirement it is not something that can be guarantied by 3rd party datasources (definitely not by old ones) so I feel we would be implementing a deduplication layer anyway in our code.